workflow.py

```diff
@@ -9,7 +9,39 @@
 def main(df: pyam.IamDataFrame) -> pyam.IamDataFrame:
     """Project/instance-specific workflow for scenario processing"""
 
-    # Run the validation and region-processing
-    dsd = DataStructureDefinition(here / "definitions")
+    dimensions = ["region", "variable"]
+    if "subannual" in df.dimensions or df.time_col == "time":
+        dimensions = dimensions + ["subannual"]
+
+    # initialize the codelists and region-processing
+    dsd = DataStructureDefinition(here / "definitions", dimensions=dimensions)
     processor = RegionProcessor.from_directory(path=here / "mappings", dsd=dsd)
-    return process(df, dsd, processor=processor)
+
+    # run the validation and region-processing
+    df = process(df, dsd, processor=processor)
```

Comment:
Would the validation fail here for anything that's not `year` in the `subannual` column?